packages/calypso-e2e/src/lib/components/support-component.ts

```diff
@@ -65,28 +67,53 @@ export class SupportComponent {
 	 */
 	async getResultCount(): Promise< { articleCount: number; linkCount: number } > {
 		return {
-			articleCount: await this.helpCenter
+			articleCount: await this.anchor
 				.getByLabel( 'Search Results' )
 				.getByRole( 'list', { name: 'Recommended resources' } )
 				.count(),
-			linkCount: await this.helpCenter
+			linkCount: await this.anchor
 				.getByLabel( 'Search Results' )
 				.getByRole( 'list', { name: 'Show me where to' } )
 				.count(),
 		};
 	}
 
 	/**
-	 * Clicks on the first result that partially matches the text.
+	 * Interact with the search result based on index.
 	 *
-	 * @param {string} text Text to match on the results.
+	 * @param {ResultsCategory} category Category of results to click on.
+	 * @param {number} index Locate results based on index. 0-indexed.
+	 */
+	async clickResultByIndex( category: ResultsCategory, index: number ) {
```

Comment:
**thought (non-blocking):** I like the flexibility that the `index` parameter gives, but I'm pondering over its necessity, especially if we're only interacting with the first result.

Reply:
> I'm pondering over its necessity, especially if we're only interacting with the first result.

Totally fair point!

I prefer to make methods applicable in a general context over singular-purpose, which is why I ended up implementing the method I did instead of something like, say, `clickFirstResult`.

This method is also a leftover from a slightly more ambitious spec I wrote then scrapped for being unworkable, where I wanted to verify that I can match on both index and title. As you can see, it didn't end up working out.

I'll change the spec to interact with something other than the first item so the method doesn't seem so out of place.

Reply:
Revised in 54fadb4fceb68c779c64ccc6f287e0daa2e7acb3